core/util/github.py

```diff
@@ -26,75 +25,46 @@ def __init__(self, q, cookie, _type='Repositories', limit=1):
+					]
+	
 	def run_crawl(self):
-		if self.type not in self.types:
-			self.type = 'Repositories'
-		urls = [f"https://{self.github}/search?q={self.q}&p={i}&type={self.type}" for i in range(1, self.limit+1)]
-		max_attempt = len(urls)
-		for url in range(max_attempt):
-			self.framework.verbose(f"[GITHUB] Searching in {url} page...")
+		max_attempt = len(self.urls)
+		for page in range(max_attempt):
 			try:
-				req = self.framework.request(url=urls[url], headers={'Cookie': self.cookie}, allow_redirects=True)
+				self.framework.verbose(f"[GITHUB] Searching in {page} page...")
+				req = self.framework.request(url=self.urls[page], headers={'Cookie': self.cookie, 'Accept': 'application/vnd.github.v3.text-match+json'}, allow_redirects=True)
+				result = req.text
+				result_json = req.json()
+				for results in result_json["items"]:
+					if 'fork' not in results:
+						self._links = self._links + "".join(results['html_url']).replace("https://", " ")
+					else:
+						self._page['Github'].append(results['html_url'])
+				repo = self._page['Github']
+				users = self._links.split(' ')
 			except:
 				self.framework.error('[GITHUB] ConnectionError')
-				max_attempt -= 1
-				if max_attempt == 0:
-					self.framework.error('Github is missed!')
-					break
-			else:
-				page = req.text
-				if f"We couldn’t find any {self.type.lower()} matching '{self.q}'</h3>" in page:
-					break
-				if f"https://github.com/search?q={self.q}&p={url+1}&type={self.type}" not in page:
-					self._pages += page
-					break 
-				self._pages += page
-	@property
-	def pages(self):
-		return self._pages
-
-	@property
-	def dns(self):
-		return self.framework.page_parse(self._pages).get_dns(self.q)
+				return
+		return
 
 	@property
-	def emails(self):
-		emails = self.page_parse(",".join(self.all_emails)).get_emails(self.q)
-		return emails
+	def links(self):
+		return self._links
 
 	@property
-	def all_emails(self):
-		parser = self.framework.page_parse(self._pages)
-		parser.pclean
-		emails = [("".join([chr(int(char[1:3], 16)) for char in email.split('&#')[1:]]))\
-					for email in parser.findall(r"(&#x\w+;)+")]
-		return list(set(emails))
+	def users(self):
+		return users
 
 	@property
 	def repositories(self):
-		parser = self.framework.page_parse(self._pages)
-		parser.pclean
-		links = [f"https://github.com/{x}" for x in parser.findall(r'data-hydro-click-hmac="[\w]+" href="([^"]+)">')]
-		links_with_desc = parser.findall(r'([\w_\./]+)</a>[\w\W]{23}class="mb-1">\s+([^\n]+)\n\s+</p>')
-		final_output = {}
-		for i in links_with_desc:
-			final_output[f"https://github.com{i[0]}"] = i[1]
-		for i in links:
-			if i not in final_output:
-				final_output[i] = "<Withoud Description>"
-		return final_output
-
-	@property
-	def users(self):
-		parser = self.framework.page_parse(self._pages)
-		parser.pclean
-		links = [f"https://github.com/{x}" for x in parser.findall(r"Unfollow ([\w_\.]+)") if x != "this"]
-		return list(set(links))
+		return repo
```

Comment:
Not defined

Reply:
Yes, my bad. I have added the changes. Thank you for pointing that out.